index.js

```diff
@@ -110,8 +110,9 @@ Prober.prototype.probe = function probe(request, bypass, callback) {
     var wrappedCallback;
     if (this.detectFailuresByCallback) {
         wrappedCallback = function(err, resp) {
-            var errResponse = resp && !isNaN(resp.statusCode) &&
-                resp.statusCode >= 500;
+            var errResponse = resp &&
+                (!isNaN(resp.statusCode) && resp.statusCode >= 500 || // HTTP
+                resp.hasOwnProperty('ok') && !resp.ok); // tchannel
```

Comment:
Also you probably do not want to mark the endpoint as `notok()` on thrift exceptions...

You probably want to just use this `probe()` as is and rely on `err`.